### PR TITLE
Created motoman_mh5_ikfast_manipulator_plugin

### PR DESCRIPTION
**Pull request by nsnitish**
IKfast plugin for motoman mh5 as a part of issue [motoman/issues/#172](https://github.com/ros-industrial/motoman/issues/172). test.cpp included in the repository tests if one of the solutions created IKfast is the true solution. 

TODO: Need to test it with moveit.

### CODE REVIEW

File: motoman_mh5_ikfast_manipulator_plugin/package.xml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  <version>0.0.0</version>
+  <description>The motoman_mh5_ikfast_manipulator_plugin package</description>
+  <!-- One maintainer tag required, multiple allowed, one person per tag -->
+  <!-- Example:  -->
+  <!-- <maintainer email="jane.doe@example.com">Jane Doe</maintainer> -->
+  <maintainer email="nitishsharma@todo.todo">nitishsharma</maintainer>
+  <!-- One license tag required, multiple allowed, one license per tag -->
+  <!-- Commonly used license strings: -->
+  <!--   BSD, MIT, Boost Software License, GPLv2, GPLv3, LGPLv2.1, LGPLv3 -->
+  <license>TODO</license>
```
Comment (jrgnicho) [inline motoman_mh5_ikfast_manipulator_plugin/package.xml:13]: I know this came from the automatic gen tool but could you set the license to BSD and clean up the file a bit?

**Review by jrgnicho — CHANGES_REQUESTED** [review]
There is a duplicate file **motoman_mh5_ikfast_manipulator_plugin/src/motoman_mh5_manipulator_ikfast_solver.cpp** in the **src/test** directory.
Also, I'm not sure why the solver code is over 20K lines of code.

**Comment by shaun-edwards** [comment]
@nsnitish, please also submit the ikfast solution to the motoman repo as well.  See [here](http://wiki.ros.org/Industrial/Tutorials/SuggestedPackageLayoutNewRepositories) for instructions, as well as the [fanuc repo](https://github.com/ros-industrial/fanuc) for an example of how to include the ikfast solver.

**Comment by shaun-edwards** [comment]
@nsnitish, I cannot find the place where you test the item requested in the linked issue.  Specifically:
> Create a test program that verifies the resulting solution returns the nearest IK solution in joint space.

Please point me to the specific code line(s) that perform this test.  As we discussed, this test should verify that the IKfast, **MoveIt** plugin (not the ikfast code itself) returns the nearest joint solution.

**Comment by gavanderhoorn** [comment]
@shaun-edwards wrote:
> @nsnitish, please also submit the ikfast solution to the motoman repo as well.  

I'd rather have this in one place. Duplication is something to be avoided.

@nsnitish: is there any specific reason why you want the plugin to be added to this repository instead of `ros-industrial/motoman_experimental`?

**Comment by shaun-edwards** [comment]
@nsnitish, now that you have addressed the smart pointer issue, we can add the test I'm interested in.  Basically, use the `getIK` and `getMultipleIK` tests as an example [here](https://github.com/ros-planning/moveit_kinematics_tests/blob/kinetic-devel/kinematics_base_test/src/test_kinematics_plugin.cpp#L355-L495).  For a Cartesian pose, call both versions of the `getPositionIK` (one returns one solution, the other returns many).  Then iterate through the multiple IK solutions and make certain that none are closer than the one, single solution.  

You will likely need to iterate through many Cartesian poses to fully test this.  

Also, this test should run fine for non-ikfast solvers since they should return the same, single answer for both calls of `getPositionIK`.

**Comment by jrgnicho** [comment]
I agree with @shaun-edwards with setting up a test that checks that the solution returned by  **getPositionIK** is indeed is the closest to the seed.  This test should apply to all plugins if done correctly (KDL, IKfast, tracIK, etc.)
I think @gavanderhoorn suggestion of avoiding duplication is valid, in fact I do think that the motoman_mh5 ikfast code should live in that robot's repo and your test code could have **moveit_kinematic_tests** as a dependency

**Comment by nsnitish** [comment]
Closing for now. Will add new PR related to above comments

**Pull request closed without merging**